Shifty.App/Components/Voucher.razor

```diff
@@ -0,0 +1,138 @@
+                            Placeholder="Available products..."
+                            AnchorOrigin="Origin.BottomCenter"
+                            ToStringFunc="@_converter"
+                            Required="true"
+                            RequiredError="Product is required"
+                            @bind-Value=_voucherForm.Product>
+
+                    @foreach (var product in _products)
+                    {
+                        <MudSelectItem Value="@product" />
+                    }
+                </MudSelect>
+
+                <MudNumericField @bind-Value="_voucherForm.Amount"
+                                Placeholder="1" 
+                                Label="Amount" 
+                                Variant="Variant.Text" 
+                                Required="true"
+                                RequiredError="Product is required" 
+                                Min="1"
+                                Max="10" />
+
+                <MudTextField T="string"
+                    @bind-Value="_voucherForm.Description"
+                    Label="Description" 
+                    Required="true"
+                    RequiredError="Description is required" />
+                
+                <MudCardActions>
+                    <MudButton  Variant="Variant.Filled"
+                                Color="Color.Primary"
+                                Class="ml-auto"
+                                Disabled="@(!_isFormValid)"
+                                OnClick="@(async () => await IssueVoucher())"
+                                EndIcon="@Icons.Material.Filled.Sell">
+                        Issue Voucher
+                    </MudButton>
+                </MudCardActions>
+            </MudForm>
+            @if (_showProgressBar)
+            {
+                <MudContainer class="d-flex justify-content-center">
+                    <MudProgressCircular hidden="@_showProgressBar" Color="Color.Default" Indeterminate="true" />
+                </MudContainer>
+            }
+            @if (_showResult)
+            {
+                <MudTextField Text="@_voucherCodes"  
+                            @ref="_multilineReference"
+                            T="string"
+                            Adornment="Adornment.End" 
+                            Style="border-width: 2px; padding: 4px;" 
+                            Outlined="true" 
+                            AdornmentIcon="@Icons.Material.Outlined.ContentCopy"
+                            OnAdornmentClick="@(async () => await CopyToClipboard())" 
+                            Lines="@_vouchers.Count" 
+                            ReadOnly=true />
+            }
+        </MudCardContent>
+    </MudCard>
+</MudContainer>
+
+@code
+{
+    private VoucherForm _voucherForm = new();
+    private bool _isFormValid = false;
+    private bool _productsLoaded = false;
+    private bool _showResult = false;
```

Comment:
Is it needed?

Reply:
Are you suggesting to do similar to the _productsLoaded and change it such that the list of tickets is set to 0 upon "Issue Vouchers" button click, and then have the amount of vouchers in the _vouchers list determine whether to show the MudTextField or not?

Or how should I interpret this comment?

Reply:
Revisited this:

`_isFormValid` is necessary to add form validation to the issue voucher button.

`_showResult` is removed and now checks `if _vouchers`is not null` and updates the same variable according to the current flow.